### PR TITLE
Pin `protobuf<=3.20.1`

### LINKED ISSUE #13159: import error with protobuf==4.21.0
## 🐛 Bug

import error when using new [protobuf](https://pypi.org/project/protobuf/#history) version

### To Reproduce

pip install pytorch_lightning protobuf

python
```
import pytorch_lightning as pl
```

Results
```
import pytorch_lightning as pl
/usr/local/lib/python3.9/site-packages/pytorch_lightning/__init__.py:30: in <module>
    from pytorch_lightning.callbacks import Callback  # noqa: E402
/usr/local/lib/python3.9/site-packages/pytorch_lightning/callbacks/__init__.py:26: in <module>
    from pytorch_lightning.callbacks.pruning import ModelPruning
/usr/local/lib/python3.9/site-packages/pytorch_lightning/callbacks/pruning.py:[31](https://github.com/openclimatefix/nowcasting_forecast/runs/6607770425?check_suite_focus=true#step:4:32): in <module>
    from pytorch_lightning.core.lightning import LightningModule
/usr/local/lib/python3.9/site-packages/pytorch_lightning/core/__init__.py:16: in <module>
    from pytorch_lightning.core.lightning import LightningModule
/usr/local/lib/python3.9/site-packages/pytorch_lightning/core/lightning.py:40: in <module>
    from pytorch_lightning.loggers import LightningLoggerBase, LoggerCollection
/usr/local/lib/python3.9/site-packages/pytorch_lightning/loggers/__init__.py:18: in <module>
    from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
/usr/local/lib/python3.9/site-packages/pytorch_lightning/loggers/tensorboard.py:26: in <module>
    from torch.utils.tensorboard import SummaryWriter
/usr/local/lib/python3.9/site-packages/torch/utils/tensorboard/__init__.py:10: in <module>
    from .writer import FileWriter, SummaryWriter  # noqa: F401
/usr/local/lib/python3.9/site-packages/torch/utils/tensorboard/writer.py:9: in <module>
    from tensorboard.compat.proto.event_pb2 import SessionLog
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/event_pb2.py:17: in <module>
    from tensorboard.compat.proto import summary_pb2 as tensorboard_dot_compat_dot_proto_dot_summary__pb2
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/summary_pb2.py:17: in <module>
    from tensorboard.compat.proto import tensor_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__pb2
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/tensor_pb2.py:16: in <module>
    from tensorboard.compat.proto import resource_handle_pb2 as tensorboard_dot_compat_dot_proto_dot_resource__handle__pb2
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/resource_handle_pb2.py:16: in <module>
    from tensorboard.compat.proto import tensor_shape_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__shape__pb2
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/tensor_shape_pb2.py:[36](https://github.com/openclimatefix/nowcasting_forecast/runs/6607770425?check_suite_focus=true#step:4:37): in <module>
    _descriptor.FieldDescriptor(
/usr/local/lib/python3.9/site-packages/google/protobuf/descriptor.py:[56](https://github.com/openclimatefix/nowcasting_forecast/runs/6607770425?check_suite_focus=true#step:4:57)0: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot not be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
```
### Expected behavior

No import error

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): pytorch-lightning-1.6.3
- PyTorch Version (e.g., 1.10): torch-1.11.0
- Python version (e.g., 3.9): 3.9
- OS (e.g., Linux): from docker image `python:3.9-slim`
- CUDA/cuDNN version:
- GPU models and configuration: CPU
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information: I also get this error on my macbook

### Additional context

Filed on [this github action](https://github.com/openclimatefix/nowcasting_forecast/runs/6607770425?check_suite_focus=true)

cc @tchaton @rohitgr7

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #13159.
Closes #13189 

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [n/a] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @tchaton @rohitgr7 @carmocca @akihironitta @borda

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Comment by akihironitta** [comment]
@Borda I have no idea why the installation test installs the latest version even though I explicitly specified `protobuf<=3.20.1` in the base requirements file. Do you have any idea?

```
...
Collecting protobuf
  Downloading protobuf-4.21.1-cp37-abi3-manylinux2014_x86_64.whl (407 kB)
...
```
https://github.com/PyTorchLightning/pytorch-lightning/runs/6639424023?check_suite_focus=true

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
> Do you have any idea?

it is coming from a dependency chain... and we install all with `-U`

File: requirements/base.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ torchmetrics>=0.4.1, <=0.7.2
 pyDeprecate>=0.3.1, <=0.3.2
 packaging>=17.0, <=21.3
 typing-extensions>=4.0.0, <4.2.1
+protobuf<=3.20.1
```
Comment (carmocca) [inline requirements/base.txt:11]:
```suggestion
protobuf<=3.20.1  TODO: remove after ...
```

**Review by carmocca — APPROVED** [review]

**Comment by akihironitta** [comment]
> I have no idea why the installation test installs the latest version even though I explicitly specified protobuf<=3.20.1 in the base requirements file. 

It turns out that the issue is with `pytorch_lightning.setup_tools._load_requirements`. Will debug it and include the patch here.

File: pytorch_lightning/setup_tools.py [hunk 2]
```diff
@@ -32,9 +32,10 @@ def _load_requirements(
     reqs = []
     for ln in lines:
         # filer all comments
+        comment = ""
         if comment_char in ln:
+            comment = ln[ln.index(comment_char) :]
             ln = ln[: ln.index(comment_char)]
-        comment = ln[ln.index(comment_char) :] if comment_char in ln else ""
```
Comment (akihironitta) [inline pytorch_lightning/setup_tools.py:37]: Fixed the issue where `comment` is always an empty string `""`.

**Comment by carmocca** [comment]
Let's also skip the quantization tests here to unblock.

**Comment by carmocca** [comment]
I'm very confused. The two GPU jobs are failing for with different errors.

(1) Lite test with DeepSpeed (cc @awaelchli)

```python
pytorch_lightning/lite/lite.py:407: in _run_impl
    return run_method(*args, **kwargs)
pytorch_lightning/lite/lite.py:412: in _run_with_strategy_setup
    return run_method(*args, **kwargs)
tests/lite/test_lite.py:402: in run
    model, optimizer = self.setup(model, optimizer)
pytorch_lightning/lite/lite.py:173: in setup
    model, optimizers = self._strategy._setup_model_and_optimizers(model, list(optimizers))
pytorch_lightning/strategies/deepspeed.py:414: in _setup_model_and_optimizers
    self.model, optimizer = self._setup_model_and_optimizer(model, optimizers[0])
pytorch_lightning/strategies/deepspeed.py:426: in _setup_model_and_optimizer
    deepspeed_engine, deepspeed_optimizer, _, _ = deepspeed.initialize(
/usr/local/lib/python3.9/dist-packages/deepspeed/__init__.py:120: in initialize
    engine = DeepSpeedEngine(args=args,
/usr/local/lib/python3.9/dist-packages/deepspeed/runtime/zero/partition_parameters.py:377: in wrapper
    if not hasattr(module, "_ds_child_entered"):
/usr/local/lib/python3.9/dist-packages/deepspeed/runtime/engine.py:432: in __getattr__
    if name in dir(self):
/usr/local/lib/python3.9/dist-packages/torch/nn/modules/module.py:1847: in __dir__
    parameters = list(self._parameters.keys())
/usr/local/lib/python3.9/dist-packages/deepspeed/runtime/engine.py:432: in __getattr__
    if name in dir(self):
/usr/local/lib/python3.9/dist-packages/torch/nn/modules/module.py:1847: in __dir__
    parameters = list(self._parameters.keys())
E   RecursionError: maximum recursion depth exceeded while calling a Python object
!!! Recursion detected (same locals & position)
=========================== short test summary info ============================
FAILED tests/lite/test_lite.py::test_deepspeed_multiple_models - RecursionErr...
```

And (2) for some reason, the test collection is collecting the `test_warning.py` file even though it's supposed to run separately. I failed to fix it with 1fdbd39.

Any help is welcome :face_with_head_bandage:

**Comment by akihironitta** [comment]
@carmocca Thank you for having looked into these flaky ~very frustrating~ issues :face_with_head_bandage:

For (2), @Borda and I experienced the same issue in the past. We don't use pytest for `test_warnings.py` anyway, so shall we add it to the ignore list so that pytest doesn't even try to collect it? (I did it in the past https://github.com/PyTorchLightning/pytorch-lightning/pull/12860/commits/cb5a0ce321519136917330e021e8b079e570d647, but the error had gone somehow so we reverted it eventually)

File: requirements/base.txt [hunk 3]
```diff
@@ -8,3 +8,4 @@ torchmetrics>=0.4.1, <=0.7.2
 pyDeprecate>=0.3.1, <=0.3.2
 packaging>=17.0, <=21.3
 typing-extensions>=4.0.0, <4.2.1
+protobuf<=3.20.1  # strict TODO: Remove after tensorboard gets compatible https://github.com/tensorflow/tensorboard/issues/5708
```
Comment (Borda) [inline requirements/base.txt:11]:
```suggestion
protobuf<=3.20.1  # strict + TODO: Remove after tensorboard gets compatible https://github.com/tensorflow/tensorboard/issues/5708
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
how about this fix: #13189

**Review by carmocca — APPROVED** [review]

**Pull request merged**